ESP Easy fails to build - fails on IRremoteESP8266

### Steps to reproduce
How can we trigger this problem?

Fresh install of Atom/Platformio on OS X 10.13.2. 
Download latest pre-release, or any pre release of ESP Easy (source code). I've tried with a number of releases, including a source tree I've built successfully last summer.
Build any target

### Expected behavior
ESP Easy should build nicely

### Actual behavior
Build fails as follows. It seems that the version of _IRremoteESP8266_ shipped with ESP Easy (1.2.0) is outdated and not compatible with the version that Platformio downloads of toolchain-xtensa (~1.40802.0). Source did compile after I've deleted _P016_IR.ino and _P035_IRTX.ino from src.

See also this [topic](http://www.letscontrolit.com/forum/viewtopic.php?f=4&t=3993&sid=330301c6e8b8514e4eb59a5e20296651).

```
lib/IRremoteESP8266/IRMitsubishiAC.cpp:80:44: note:   mismatched types 'std::initializer_list<_Tp>' and 'unsigned int'
temp = max(MITSUBISHI_AC_MIN_TEMP, temp);
^
In file included from /Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++
/4.8.2/algorithm:61:0,
from /Users/sander/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:240,

from lib/IRremoteESP8266/IRremoteInt.h:33,
from lib/IRremoteESP8266/IRremoteESP8266.h:35,
from lib/IRremoteESP8266/IRMitsubishiAC.h:2,
from lib/IRremoteESP8266/IRMitsubishiAC.cpp:12:
/Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algoba
se.h:260:5: note: template<class _Tp, class _Compare> const _Tp& std::max(const _Tp&, const _Tp&, _Com
pare)
max(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
/Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algoba
se.h:260:5: note:   template argument deduction/substitution failed:
lib/IRremoteESP8266/IRMitsubishiAC.cpp:80:44: note:   deduced conflicting types for parameter 'const _

Tp' ('unsigned int' and 'uint8_t {aka unsigned char}')
temp = max(MITSUBISHI_AC_MIN_TEMP, temp);
^
In file included from /Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++
/4.8.2/algorithm:61:0,
from /Users/sander/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:240,
from lib/IRremoteESP8266/IRremoteInt.h:33,
from lib/IRremoteESP8266/IRremoteESP8266.h:35,
from lib/IRremoteESP8266/IRMitsubishiAC.h:2,
from lib/IRremoteESP8266/IRMitsubishiAC.cpp:12:
/Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algoba
se.h:216:5: note: template<class _Tp> const _Tp& std::max(const _Tp&, const _Tp&)
max(const _Tp& __a, const _Tp& __b)
^
/Users/sander/.platformio/packages/toolchain-xtensa/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algoba
se.h:216:5: note:   template argument deduction/substitution failed:
lib/IRremoteESP8266/IRMitsubishiAC.cpp:80:44: note:   deduced conflicting types for parameter 'const _
Tp' ('unsigned int' and 'uint8_t {aka unsigned char}')
temp = max(MITSUBISHI_AC_MIN_TEMP, temp);
^
lib/IRremoteESP8266/IRMitsubishiAC.cpp:81:44: error: no matching function for call to 'min(unsigned in
t, uint8_t&)'
```

### System configuration
Not relevant

Software or git version:
https://github.com/letscontrolit/ESPEasy/releases/tag/mega-20180114

--- a/lib/IRremoteESP8266/IRKelvinator.cpp
+++ b/lib/IRremoteESP8266/IRKelvinator.cpp
@@ -93,8 +93,8 @@ bool ICACHE_FLASH_ATTR IRKelvinatorAC::getPower() {
 
 // Set the temp. in deg C
 void ICACHE_FLASH_ATTR IRKelvinatorAC::setTemp(uint8_t temp) {
-    temp = max(KELVINATOR_MIN_TEMP, temp);
-    temp = min(KELVINATOR_MAX_TEMP, temp);
+    temp = max(static_cast<uint8_t>(KELVINATOR_MIN_TEMP), temp);
+    temp = min(static_cast<uint8_t>(KELVINATOR_MAX_TEMP), temp);
     remote_state[1] = (remote_state[1] & 0xF0U) | (temp - KELVINATOR_MIN_TEMP);
     remote_state[9] = remote_state[1];  // Duplicate to the 2nd command chunk.
 }
@@ -106,12 +106,12 @@ uint8_t ICACHE_FLASH_ATTR IRKelvinatorAC::getTemp() {
 
 // Set the speed of the fan, 0-5, 0 is auto, 1-5 is the speed
 void ICACHE_FLASH_ATTR IRKelvinatorAC::setFan(uint8_t fan) {
-  fan = min(KELVINATOR_FAN_MAX, fan);  // Bounds check
+  fan = min(static_cast<uint8_t>(KELVINATOR_FAN_MAX), fan);  // Bounds check
 
   // Only change things if we need to.
   if (fan != getFan()) {
     // Set the basic fan values.
-    uint8_t fan_basic = min(KELVINATOR_BASIC_FAN_MAX, fan);
+    uint8_t fan_basic = min(static_cast<uint8_t>(KELVINATOR_BASIC_FAN_MAX), fan);
     remote_state[0] = (remote_state[0] & KELVINATOR_BASIC_FAN_MASK) |
         (fan_basic << KELVINATOR_FAN_OFFSET);
     remote_state[8] = remote_state[0];  // Duplicate to the 2nd command chunk.

--- a/lib/IRremoteESP8266/IRMitsubishiAC.cpp
+++ b/lib/IRremoteESP8266/IRMitsubishiAC.cpp
@@ -77,8 +77,8 @@ bool ICACHE_FLASH_ATTR IRMitsubishiAC::getPower() {
 
 // Set the temp. in deg C
 void ICACHE_FLASH_ATTR IRMitsubishiAC::setTemp(uint8_t temp) {
-    temp = max(MITSUBISHI_AC_MIN_TEMP, temp);
-    temp = min(MITSUBISHI_AC_MAX_TEMP, temp);
+    temp = max(static_cast<uint8_t>(MITSUBISHI_AC_MIN_TEMP), temp);
+    temp = min(static_cast<uint8_t>(MITSUBISHI_AC_MAX_TEMP), temp);
     remote_state[7] = temp - MITSUBISHI_AC_MIN_TEMP;
 }
 
@@ -136,7 +136,7 @@ void ICACHE_FLASH_ATTR IRMitsubishiAC::setMode(uint8_t mode) {
 
 // Set the requested vane operation mode of the a/c unit.
 void ICACHE_FLASH_ATTR IRMitsubishiAC::setVane(uint8_t mode) {
-  mode = max(mode, B111);  // bounds check
+  mode = max(mode, static_cast<uint8_t>(B111));  // bounds check
   mode |= B1000;
   mode <<= 3;
   remote_state[9] |= mode;

--- a/lib/IRremoteESP8266/IRremoteESP8266.cpp
+++ b/lib/IRremoteESP8266/IRremoteESP8266.cpp
@@ -155,7 +155,7 @@ void ICACHE_FLASH_ATTR IRsend::sendNEC (unsigned long data, int nbits,
   // Footer
   mark(NEC_BIT_MARK);
   // Gap to next command.
-  space(max(0, NEC_MIN_COMMAND_LENGTH - usecs.elapsed()));
+  space(max(0ul, NEC_MIN_COMMAND_LENGTH - usecs.elapsed()));
 
   // Optional command repeat sequence.
   for (unsigned int i = 0; i < repeat; i++) {
@@ -164,7 +164,7 @@ void ICACHE_FLASH_ATTR IRsend::sendNEC (unsigned long data, int nbits,
     space(NEC_RPT_SPACE);
     mark(NEC_BIT_MARK);
     // Gap till next command.
-    space(max(0, NEC_MIN_COMMAND_LENGTH - usecs.elapsed()));
+    space(max(0ul, NEC_MIN_COMMAND_LENGTH - usecs.elapsed()));
   }
 }
 
@@ -236,7 +236,7 @@ void ICACHE_FLASH_ATTR IRsend::sendSony(unsigned long data, int nbits,
     // Footer
     // The Sony protocol requires us to wait 45ms from start of a code to the
     // start of the next one. A 10ms minimum gap is also required.
-    space(max(10000, 45000 - usecs.elapsed()));
+    space(max(10000u, 45000 - usecs.elapsed()));
   }
   // A space() is always performed last, so no need to turn off the LED.
 }
@@ -367,7 +367,7 @@ void ICACHE_FLASH_ATTR IRsend::sendRCMM(uint32_t data, uint8_t nbits) {
   mark(RCMM_BIT_MARK);
   // Protocol requires us to wait at least RCMM_RPT_LENGTH usecs from the start
   // or RCMM_MIN_GAP usecs.
-  space(max(RCMM_RPT_LENGTH - usecs.elapsed(), RCMM_MIN_GAP));
+  space(max(RCMM_RPT_LENGTH - usecs.elapsed(), static_cast<uint32_t>(RCMM_MIN_GAP)));
 }
 
 void ICACHE_FLASH_ATTR IRsend::sendPanasonic(unsigned int address,
@@ -415,7 +415,7 @@ void ICACHE_FLASH_ATTR IRsend::sendJVC(unsigned long data, int nbits,
     // Footer
     mark(JVC_BIT_MARK);
     // Wait till the end of the repeat time window before we send another code.
-    space(max(0, JVC_RPT_LENGTH - usecs.elapsed()));
+    space(max(0u, JVC_RPT_LENGTH - usecs.elapsed()));
     usecs.reset();
   }
   // No need to turn off the LED as we will always end with a space().
@@ -647,7 +647,7 @@ void ICACHE_FLASH_ATTR IRsend::sendSherwood(unsigned long data, int nbits,
                                             unsigned int repeat) {
   // Sherwood remote codes appear to be NEC codes with a manditory repeat code.
   // i.e. repeat should be >= 1.
-  sendNEC(data, nbits, max(1, repeat));
+  sendNEC(data, nbits, max(1u, repeat));
 }
 
 void ICACHE_FLASH_ATTR IRsend::sendMitsubishiAC(unsigned char data[]) {
@@ -911,7 +911,7 @@ bool ICACHE_FLASH_ATTR IRrecv::decode(decode_results *results,
 //   Nr. of ticks.
 uint32_t IRrecv::ticksLow(uint32_t usecs, uint8_t tolerance) {
   // max() used to ensure the result can't drop below 0 before the cast.
-  return((uint32_t) max(usecs * (1.0 - tolerance/100.)/USECPERTICK, 0));
+  return((uint32_t) max(usecs * (1.0 - tolerance/100.)/USECPERTICK, 0.0));
 }
 
 // Calculate the upper bound of the nr. of ticks.

--- a/src/Misc.ino
+++ b/src/Misc.ino
@@ -733,7 +733,7 @@ boolean str2ip(const char *string, byte* IP)
 
 /********************************************************************************************\
   check the program memory hash
-  The const MD5_MD5_MD5_MD5_BoundariesOfTheSegmentsGoHere... needs to remain unchanged as it will be replaced by 
+  The const MD5_MD5_MD5_MD5_BoundariesOfTheSegmentsGoHere... needs to remain unchanged as it will be replaced by
   - 16 bytes md5 hash, followed by
   - 4 * uint32_t start of memory segment 1-4
   - 4 * uint32_t end of memory segment 1-4
@@ -741,24 +741,27 @@ boolean str2ip(const char *string, byte* IP)
   Execution time 520kb @80Mhz: 236ms
   Returns: 0 if hash compare fails, number of checked bytes otherwise.
   The reference hash is calculated by a .py file and injected into the binary.
-  Caution: currently the hash sits in an unchecked segment. If it ever moves to a checked segment, make sure 
-  it is excluded from the calculation !    
+  Caution: currently the hash sits in an unchecked segment. If it ever moves to a checked segment, make sure
+  it is excluded from the calculation !
   \*********************************************************************************************/
-void dump (uint32_t addr){
-       Serial.print (addr,HEX);
-       Serial.print(": ");
-       for (uint32_t a = addr; a<addr+16; a++)
-          {  
-          Serial.print  (  pgm_read_byte(a),HEX);
-          Serial.print (" ");
-          }
-       Serial.println("");   
-} 
+#if defined(ARDUINO_ESP8266_RELEASE_2_3_0)
+void dump (uint32_t addr) { //Seems already included in core 2.4 ...
+  Serial.print (addr, HEX);
+  Serial.print(": ");
+  for (uint32_t a = addr; a < addr + 16; a++)
+  {
+    Serial.print ( pgm_read_byte(a), HEX);
+    Serial.print (" ");
+  }
+  Serial.println("");
+}
+#endif
+
 uint32_t progMemMD5check(){
-    #define BufSize 10     
+    #define BufSize 10
     const  char CRCdummy[16+32+1] = "MD5_MD5_MD5_MD5_BoundariesOfTheSegmentsGoHere...";                   // 16Bytes MD5, 32 Bytes Segment boundaries, 1Byte 0-termination. DO NOT MODIFY !
-    uint32_t calcBuffer[BufSize];         
-    uint32_t md5NoOfBytes = 0; 
+    uint32_t calcBuffer[BufSize];
+    uint32_t md5NoOfBytes = 0;
     memcpy (calcBuffer,CRCdummy,16);                                                                      // is there still the dummy in memory ? - the dummy needs to be replaced by the real md5 after linking.
     if( memcmp (calcBuffer, "MD5_MD5_MD5_MD5_",16)==0){                                                   // do not memcmp with CRCdummy directly or it will get optimized away.
         addLog(LOG_LEVEL_INFO, F("CRC  : No program memory checksum found. Check output of crc2.py"));
@@ -767,12 +770,12 @@ uint32_t progMemMD5check(){
     MD5Builder md5;
     md5.begin();
     for (int l = 0; l<4; l++){                                                                            // check max segments,  if the pointer is not 0
-        uint32_t *ptrStart = (uint32_t *)&CRCdummy[16+l*4]; 
-        uint32_t *ptrEnd =   (uint32_t *)&CRCdummy[16+4*4+l*4]; 
+        uint32_t *ptrStart = (uint32_t *)&CRCdummy[16+l*4];
+        uint32_t *ptrEnd =   (uint32_t *)&CRCdummy[16+4*4+l*4];
         if ((*ptrStart) == 0) break;                                                                      // segment not used.
-        for (uint32_t i = *ptrStart; i< (*ptrEnd) ; i=i+sizeof(calcBuffer)){                              // "<" includes last byte 
+        for (uint32_t i = *ptrStart; i< (*ptrEnd) ; i=i+sizeof(calcBuffer)){                              // "<" includes last byte
              for (int buf = 0; buf < BufSize; buf ++){
-                calcBuffer[buf] = pgm_read_dword((void*)i+buf*4);                                         // read 4 bytes
+                calcBuffer[buf] = pgm_read_dword((uint32_t*)i+buf*4);                                         // read 4 bytes
                 md5NoOfBytes+=sizeof(calcBuffer[0]);
              }
              md5.add((uint8_t *)&calcBuffer[0],(*ptrEnd-i)<sizeof(calcBuffer) ? (*ptrEnd-i):sizeof(calcBuffer) );     // add buffer to md5. At the end not the whole buffer. md5 ptr to data in ram.
@@ -785,11 +788,11 @@ uint32_t progMemMD5check(){
       return md5NoOfBytes;
    }
    addLog(LOG_LEVEL_INFO, F("CRC  : program checksum     ...FAIL"));
-   return 0; 
+   return 0;
 }
 
 
-  
+
 /********************************************************************************************\
   Save settings to SPIFFS
   \*********************************************************************************************/
@@ -801,26 +804,26 @@ String SaveSettings(void)
   md5.add((uint8_t *)&Settings, sizeof(Settings)-16);
   md5.calculate();
   md5.getBytes(Settings.md5);
-  
+
   String err;
   err=SaveToFile((char*)FILE_CONFIG, 0, (byte*)&Settings, sizeof(struct SettingsStruct));
   if (err.length())
    return(err);
 
-  
+
   memcpy( SecuritySettings.ProgmemMd5, thisBinaryMd5, 16);
   md5.begin();
   md5.add((uint8_t *)&SecuritySettings, sizeof(SecuritySettings)-16);
   md5.calculate();
   md5.getBytes(SecuritySettings.md5);
   err=SaveToFile((char*)FILE_SECURITY, 0, (byte*)&SecuritySettings, sizeof(struct SecurityStruct));
-  
+
 //  dump((uint32_t)SecuritySettings.md5);
 //  dump((uint32_t)SecuritySettings.ProgmemMd5);
 //  dump((uint32_t)Settings.md5);
 //  dump((uint32_t)Settings.ProgmemMd5);
 //  dump((uint32_t)thisBinaryMd5);
-    
+
  return (err);
 }
 
@@ -832,7 +835,7 @@ String LoadSettings()
   String err;
   uint8_t calculatedMd5[16];
   MD5Builder md5;
-  
+
   err=LoadFromFile((char*)FILE_CONFIG, 0, (byte*)&Settings, sizeof( SettingsStruct));
   if (err.length())
     return(err);
@@ -843,13 +846,13 @@ String LoadSettings()
   md5.getBytes(calculatedMd5);
   if (memcmp (calculatedMd5, Settings.md5,16)==0){
     addLog(LOG_LEVEL_INFO,  F("CRC  : Settings CRC           ...OK"));
-    if (memcmp(Settings.ProgmemMd5, thisBinaryMd5, 16)!=0) 
-      addLog(LOG_LEVEL_INFO, F("CRC  : binary has changed since last save of Settings"));  
+    if (memcmp(Settings.ProgmemMd5, thisBinaryMd5, 16)!=0)
+      addLog(LOG_LEVEL_INFO, F("CRC  : binary has changed since last save of Settings"));
   }
   else{
     addLog(LOG_LEVEL_ERROR, F("CRC  : Settings CRC           ...FAIL"));
   }
-  
+
 
   err=LoadFromFile((char*)FILE_SECURITY, 0, (byte*)&SecuritySettings, sizeof( SecurityStruct));
   md5.begin();
@@ -858,8 +861,8 @@ String LoadSettings()
   md5.getBytes(calculatedMd5);
   if (memcmp (calculatedMd5, SecuritySettings.md5, 16)==0){
     addLog(LOG_LEVEL_INFO, F("CRC  : SecuritySettings CRC   ...OK "));
-    if (memcmp(SecuritySettings.ProgmemMd5,thisBinaryMd5, 16)!=0) 
-      addLog(LOG_LEVEL_INFO, F("CRC  : binary has changed since last save of Settings"));  
+    if (memcmp(SecuritySettings.ProgmemMd5,thisBinaryMd5, 16)!=0)
+      addLog(LOG_LEVEL_INFO, F("CRC  : binary has changed since last save of Settings"));
  }
   else{
     addLog(LOG_LEVEL_ERROR, F("CRC  : SecuritySettings CRC   ...FAIL"));
@@ -1507,60 +1510,6 @@ uint32_t getChecksum(byte* buffer, size_t size)
   return sum;
 }
 
-
-String getDayString()
-{
-  String reply;
-  if (day() < 10)
-    reply += F("0");
-  reply += day();
-  return reply;
-}
-String getMonthString()
-{
-  String reply;
-  if (month() < 10)
-    reply += F("0");
-  reply += month();
-  return reply;
-}
-String getYearString()
-{
-  String reply = String(year());
-  return reply;
-}
-String getYearStringShort()
-{
-  String dummy = String(year());
-  String reply = dummy.substring(2);
-  return reply;
-}
-
-
-String getHourString()
-{
-  String reply;
-  if (hour() < 10)
-    reply += F("0");
-  reply += String(hour());
-  return reply;
-}
-String getMinuteString()
-{
-  String reply;
-  if (minute() < 10)
-    reply += F("0");
-  reply += minute();
-  return reply;
-}
-String getSecondString()
-{
-  String reply;
-  if (second() < 10)
-    reply += F("0");
-  reply += second();
-  return reply;
-}
 
 
 /********************************************************************************************\

